clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntity.java

```diff
@@ -127,4 +127,27 @@ private EhcacheEntityResponse invoke(EhcacheEntityMessage message) throws Except
       }
     }
   }
+
+  public EhcacheEntityResponse invokeCompleted(EhcacheEntityMessage message) throws Exception {
+    InvokeFuture<EhcacheEntityResponse> result = endpoint.beginInvoke().message(message).ackCompleted().invoke();
+    boolean interrupted = false;
+    try {
+      while (true) {
+        try {
+          EhcacheEntityResponse response = result.get();
+          if (Type.FAILURE.equals(response.getType())) {
+            throw ((Failure) response).getCause();
```

Comment:
Thats nice... but what if the cause of the remote operation failure is an InterruptedException... the loop will never terminate.  You need to remove the response processing outside the try catch don't you?